frontend/src/components/modals/settings/SettingsModal.tsx

```diff
@@ -59,17 +61,25 @@ function SettingsModal({ isOpen, onOpenChange }: SettingsProps) {
   };
 
   const handleLanguageChange = (language: string) => {
-    const key = AvailableLanguages.find(
-      (lang) => lang.label === language,
-    )?.value;
-
-    if (key) setSettings((prev) => ({ ...prev, LANGUAGE: key }));
+    const key =
+      AvailableLanguages.find((lang) => lang.label === language)?.value ||
+      language;
+    // The appropriate key is assigned when the user selects a language.
+    // Otherwise, their input is reflected in the inputValue field of the Autocomplete component.
+    setSettings((prev) => ({ ...prev, LANGUAGE: key }));
   };
 
   const handleAPIKeyChange = (key: string) => {
     setSettings((prev) => ({ ...prev, LLM_API_KEY: key }));
   };
 
+  const handleResetSettings = () => {
```

Comment:
These handlers all edit the `settings` state. Maybe it is better to directly modify the `settings` state rather than call these handlers and change state four times. E.g.,

```ts
const handleResetSettings = () => {
  setSettings(getDefaultSettings());
}
```